### PR TITLE
Fixed "Fine-Tune Prompt" link not working on "Compare Run Methods" page

### PR DESCRIPTION
**Pull request by aiyakitori**
## What does this PR do?

Previously in "Compare Run Methods" page tapping on "Fine-Tune Prompt" does not navigate to the corresponding Fine Tune run. Added logic to construct url for fine tune prompt ids. 

## Checklists

- [YES] Tests have been run locally and passed
- [YES] New tests have been added to any work in /lib

https://github.com/user-attachments/assets/b45dc179-918b-43b4-ae5e-a5d0b785b632

## Summary by CodeRabbit

* **New Features**
  * Improved handling of prompt links for fine-tuned prompts, generating more accurate URLs for prompt IDs with the "fine_tune_prompt::" prefix.

* **Tests**
  * Added comprehensive tests to verify correct URL construction for various prompt ID formats and edge cases.

### CODE REVIEW

File: app/web_ui/src/lib/utils/link_builder.ts [hunk 1]
```diff
@@ -6,6 +8,14 @@ export function prompt_link(
   if (!project_id || !task_id || !prompt_id) {
     return undefined
   }
+  // Special case for fine-tuned prompts
+  if (prompt_id.startsWith(FINE_TUNE_PROMPT_PREFIX)) {
+    // Get the last component of the prompt ID. fine_tune_prompt::fine_tune_id
+    const trimmed_prompt_id = prompt_id.replace(FINE_TUNE_PROMPT_PREFIX, "")
+    const fine_tune_id =
+      trimmed_prompt_id.split("::").pop() || trimmed_prompt_id
+    return `/fine_tune/${project_id}/${task_id}/fine_tune/${encodeURIComponent(fine_tune_id)}`
```
Comment (scosman) [inline app/web_ui/src/lib/utils/link_builder.ts:17]: nit: why encode one but not the others? I'm fine either way (I think our IDs only allow url compatible characters so fine without), but should be consistent.

File: app/web_ui/src/lib/utils/link_builder.ts [hunk 2]
```diff
@@ -6,6 +8,14 @@ export function prompt_link(
   if (!project_id || !task_id || !prompt_id) {
     return undefined
   }
+  // Special case for fine-tuned prompts
+  if (prompt_id.startsWith(FINE_TUNE_PROMPT_PREFIX)) {
+    // Get the last component of the prompt ID. fine_tune_prompt::fine_tune_id
+    const trimmed_prompt_id = prompt_id.replace(FINE_TUNE_PROMPT_PREFIX, "")
+    const fine_tune_id =
```
Comment (scosman) [inline app/web_ui/src/lib/utils/link_builder.ts:15]: why the split/pop? If the format is `fine_tune_prompt::fine_tune_id` then replace should do it.

File: app/web_ui/src/lib/utils/link_builder.ts [hunk 3]
```diff
@@ -6,6 +8,14 @@ export function prompt_link(
   if (!project_id || !task_id || !prompt_id) {
     return undefined
   }
+  // Special case for fine-tuned prompts
```
Comment (scosman) [inline app/web_ui/src/lib/utils/link_builder.ts:11]: what about task_run_config::? I guess we don't have a page for them?

Comment (aiyakitori) [reply app/web_ui/src/lib/utils/link_builder.ts:11]: It seems like it, they are listed under "Compare Run Methods" page.

Comment (scosman) [reply app/web_ui/src/lib/utils/link_builder.ts:11]: ah, please link it!

Comment (scosman) [reply app/web_ui/src/lib/utils/link_builder.ts:11]: oh, but that needs and eval ID? We need to add a screen for these eventually. TBD when/where.

File: app/web_ui/src/lib/utils/link_builder.ts (hunk 2 above)
Comment (aiyakitori) [reply app/web_ui/src/lib/utils/link_builder.ts:15]: Turns out the format of the id is `fine_tune_prompt::[project_id]::[task_id]::fine_tune_id`

File: app/web_ui/src/lib/utils/link_builder.ts (hunk 1 above)
Comment (aiyakitori) [reply app/web_ui/src/lib/utils/link_builder.ts:17]: This is following the same pattern as other links. It only encodes the last component.

**Review by scosman — APPROVED** [review]
